### ISSUE TITLE
Embed Service Request: Tiktok

### ISSUE THREAD

**Issue by MrT-tak** [StarCitizenWiki/mediawiki-extensions-EmbedVideo#106]
**Service Name**
TikTok

**URL to an example embed**
[A link to example content that would be embedded by this extension: https://www.tiktok.com/@washiezonroblox/video/7406477724249804074](https://www.tiktok.com/@washiezonroblox/video/7406477724249804074)

I am trying to embed TikTok because the PR team of a Roblox game created it and it is great for how to tutorial in my wiki(specializing in that said game on Roblox)

**Additional notes**
I am trying to embed TikTok because the PR team of a Roblox game created it and it is great for how to tutorial in my wiki(specializing in that said game on Roblox)

**This is the developer documentation of TikTok embedding videos. It has APIs. https://developers.tiktok.com/doc/embed-videos**

Sadly, I would agree with you that TikTok has some serious privacy allegations.

Thanks!

**Comment by MrT-tak** [comment]
Response

The response format follows the specification of https://oembed.com/, please check it out for more details.

Example
Request URL

https://www.tiktok.com/oembed?url=https://www.tiktok.com/@scout2015/video/6718335390845095173
Response Data

{
  "version": "1.0",
  "type": "video",
  "title": "Scramble up ur name & I’ll try to guess it😍❤️ #foryoupage #petsoftiktok #aesthetic",
  "author_url": "https://www.tiktok.com/@scout2015",
  "author_name": "Scout & Suki",
  "width": "100%",
  "height": "100%",
  "html": "<blockquote class=\"tiktok-embed\" cite=\"https://www.tiktok.com/@scout2015/video/6718335390845095173\" data-video-id=\"6718335390845095173\" data-embed-from=\"oembed\" style=\"max-width: 605px;min-width: 325px;\" > <section> <a target=\"_blank\" title=\"@scout2015\" href=\"https://www.tiktok.com/@scout2015?refer=embed\">@scout2015</a> <p>Scramble up ur name & I’ll try to guess it😍❤️ <a title=\"foryoupage\" target=\"_blank\" href=\"https://www.tiktok.com/tag/foryoupage?refer=embed\">#foryoupage</a> <a title=\"petsoftiktok\" target=\"_blank\" href=\"https://www.tiktok.com/tag/petsoftiktok?refer=embed\">#petsoftiktok</a> <a title=\"aesthetic\" target=\"_blank\" href=\"https://www.tiktok.com/tag/aesthetic?refer=embed\">#aesthetic</a></p> <a target=\"_blank\" title=\"♬ original sound - 𝐇𝐚𝐰𝐚𝐢𝐢𓆉\" href=\"https://www.tiktok.com/music/original-sound-6689804660171082501?refer=embed\">♬ original sound - 𝐇𝐚𝐰𝐚𝐢𝐢𓆉</a> </section> </blockquote> <script async src=\"https://www.tiktok.com/embed.js\"></script>",
  "thumbnail_width": 720,
  "thumbnail_height": 1280,
  "thumbnail_url": "https://p16.muscdn.com/obj/tos-maliva-p-0068/06kv6rfcesljdjr45ukb0000d844090v0200010605",
  "provider_url": "https://www.tiktok.com",
  "provider_name": "TikTok"
}

**Comment by octfx** [comment]
Looking at the documentation, it seems like that TikTok does not use traditional iframes for embedding video, but rather rolls their own implementation by loading `https://www.tiktok.com/embed.js` for each embed.

EmbedVideo assumes that embeds are handled through iframes, leading to tiktok not being easily integrated.

I currently do not have the time available to expand EV such that arbitrary embeds are possible. :/
I am more than welcome to accept and review PRs though.

**Comment by MrT-tak** [comment]
Thank you very much!
I tried just putting the non-iframe cde on my page but it didn't work.
Thanks for checking this issue, I will try to understand how this extension work.

**Comment by Daniels5** [comment]
Have you ever tried any of this as follow？

`<iframe src="https://www.tiktok.com/embed/7406477724249804074/" style="width:323px; height:700px;" sandbox="allow-scripts  allow-same-origin allow-popups" loading="lazy" referrerpolicy="no-referrer" frameborder="0" scrolling="no" allow="fullscreen" autoplay="0"></iframe>`

**Comment by MrT-tak** [comment]
I presume someone sent a solution that may not have worked and thus deleted it. Thanks for the attempt!

**Comment by MrT-tak** [comment]
After viewing the potentially deleted solution, there is a URL which is responsible for embeds. 
https://www.tiktok.com/embed/[
video_id]/

such as https://www.tiktok.com/embed/7406477724249804074/.

Before pinging octfx, let me try this page...

**Comment by MrT-tak** [comment]
After viewing the potentially deleted solution, there is a URL which is responsible for embeds. https://www.tiktok.com/embed/[
video_id]/

@octfx Any thoughts? (sorry for the ping) I tried entering the URL for this but it couldn't parse (because there is not support) Please add a way to add https://www.tiktok.com/video/@(creator)/(id) and show a iframe of https://www.tiktok.com/embed/(id)/

**Comment by MrT-tak** [comment]
directly putting html iframe code to the wiki didn't work... 😔

**Comment by octfx** [comment]
@MrT-tak you can temporarily set `$wgRawHtml=true;`  
See: https://www.mediawiki.org/wiki/Manual:$wgRawHtml

**Comment by MrT-tak** [comment]
I'm really not sure.. I'll wait for official implementation. I use fandom which is complicating things even more.

**Comment by MrT-tak** [comment]
I've revisited the repo's code. Although, my experience in Javascripts allows me to understand basic things like const but I have no idea what is going on.

**Comment by MrT-tak** [comment]
To people looking to contribute,
So far we found https://www.tiktok.com/ (Video's id) which is useful for embedding. However this isn' tdocumented.
We just need a pull request using this url.

**Comment by MrT-tak** [comment]
> I've revisited the repo's code. Although, my experience in Javascripts allows me to understand basic things like const but I have no idea what is going on.

Yeah, I forked it seeing hope and currage in my self but I lost confidence as soon as I saw PHP. I made a fork but I gave up lol, if you know PHP and js it may be easy for you but personally, I wont make a  pr

**Comment by Dianliang233** [comment]
It might be easier to solve this by creating an api route that resolves the oembed into an html document and then in the article page iframe the api route?
